### ISSUE TITLE
No blank lines between import groups, please

### ISSUE THREAD

**Issue by JordanLongstaff** [daidodo/tsimportsorter#3]
I would like to have an option to disable the blank lines that get inserted between import groups.

**Comment by daidodo** [comment]
Hi @JordanLongstaff,

May I ask what you're trying to achieve? If you don't want grouping, you can set `groupRules: []` then all will be in the default group.
Or do you want a customisable sorting rule inside a group?
Thanks!

**Comment by JordanLongstaff** [comment]
If I set `groupRules: []` then they all go in the default group, but they get organized wrong. It makes relative imports get moved above absolute imports.

**Comment by JordanLongstaff** [comment]
The point is, I want them to be grouped like they have been, but without being separated by blank lines.

**Comment by JordanLongstaff** [comment]
Can I construct a group rule that puts relative imports and absolute imports in the same group, but still puts all the relative imports last?

**Comment by JordanLongstaff** [comment]
Just FYI, the motivation here is consistency at my workplace. We also use Python and Python has special rules for how imports are ordered.

**Comment by daidodo** [comment]
Thanks for your clarification!
I can think of a few options of the solution:
#### `leadingEmptyLines` and `trailingEmptyLines`
```json
groupRules: [
  { ... },
  {
    "regex": "...",
    "leadingEmptyLines": 0,
    "trailingEmptyLines": 0,
  },
  { ... }
]
```
This is a mimic by removing blank lines between groups as you mentioned. But there are a few edge cases, e.g. what if the first group in the file sets `leadingEmptyLines: 0`:
```ts
/* LICENSE
 * ...
 */
import A from './a';  // Oops, global comments becomes statement comments.
...
``` 
Similar problems occur to `trailingEmptyLines` of the last group.
#### Customisable sorting rule within a group
In your example, you probably need a rule like `[a-zA-Z]` < `[.]`. But it'll be non-trivial to reproduce the same order as the above mimic option.
#### Sub-groups
```json
groupRules: [
  { ... },
  {
    "regex": ["^[^@.]", "^[@]", "^[.]"]
  },
  { ... }
]
```
By extending `regex` to be either a string or an array of strings, we can sub-group imports and sort as we want. I guess this is the closest to what you really want. The question would be how deep (and complex) we want to go, e.g. can we specify sorting rules for a sub-group.

Please tell me your thoughts or any other options I'm missing.
Thanks!

**Comment by JordanLongstaff** [comment]
I can't think of any other options you'd be missing. If I had to choose between the options you've mentioned, I think I like the subgroups best.

**Comment by daidodo** [comment]
Resolved in v1.2.0.
